cmd/app/http_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package app
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/identity"
+	"github.com/spf13/viper"
+)
+
+func setupHTTPServer(t *testing.T) (httpServer, string) {
+	t.Helper()
+	httpListen, _ := net.Listen("tcp", ":0")
+
+	viper.Set("grpc-host", "")
+	viper.Set("grpc-port", 0)
+	grpcServer, err := createGRPCServer(nil, nil, &TrivialCertificateAuthority{})
+	if err != nil {
+		t.Error(err)
+	}
+	grpcServer.startTCPListener()
+	// loop until server starts listening in separate goroutine
+	for {
```

Comment:
nit: Would it be worth adding a timeout (https://gobyexample.com/timeouts) to avoid this test spinning for awhile? Not sure what the default timeout is for a test.